app/src/main/java/com/mad41/tripreminder/HistoryFragment.java

```diff
@@ -0,0 +1,134 @@
+import androidx.recyclerview.widget.RecyclerView;
+
+import android.os.Handler;
+import android.os.Message;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
+
+import com.mad41.tripreminder.room_database.MyRoomDataBase;
+import com.mad41.tripreminder.room_database.trip.Trip;
+import com.mad41.tripreminder.room_database.view_model.TripViewModel;
+import com.mad41.tripreminder.trip_ui.HistoryAdapter;
+import com.mad41.tripreminder.trip_ui.NoteReviewDialogue;
+
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class HistoryFragment extends Fragment {
+    RecyclerView recyclerView;
+    HistoryAdapter adapter;
+    private TripViewModel tripViewModel;
+    RecyclerView.LayoutManager layoutManager;
+    List<Trip> tripModelArrayList;
+    historyListner listner;
+    int ID;
+
+    public HistoryFragment() {
+        // Required empty public constructor
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        View fragment =  inflater.inflate(R.layout.fragment_history, container, false);
+
+        recyclerView = (RecyclerView) fragment.findViewById(R.id.HistoryRecyclerView);
+        tripModelArrayList = new ArrayList<>();
+        recyclerView.setHasFixedSize(true);
+        layoutManager = new LinearLayoutManager(getContext().getApplicationContext());
+        LinearLayoutManager layoutManager = new LinearLayoutManager(getContext().getApplicationContext());
+        layoutManager.setOrientation(RecyclerView.VERTICAL);
+        recyclerView.setLayoutManager(layoutManager);
+        tripViewModel = ViewModelProviders.of(requireActivity()).get(TripViewModel.class);
+        tripViewModel.getHistoryNotes().observe(requireActivity(), new Observer<List<Trip>>() {
+            @Override
+            public void onChanged(List<Trip> trips) {
+                tripModelArrayList=trips;
+                listner = new historyListner() {
+                    @Override
+                    public void showNotes(List<Trip> notes , int id) {
+
+                        Trip trip = null;
+                        for (int i = 0; i < notes.size(); i++) {
+                            if (notes.get(i).getId() == id) {
+                                trip = notes.get(i);
+                            }
+                        }
+                        NoteReviewDialogue noteReviewDialogue = new NoteReviewDialogue(trip);
+                        noteReviewDialogue.show(getActivity().getSupportFragmentManager(), "frag");
+
+                    }
+
+                    @Override
+                    public void DeleteTrip(List<Trip> Trips, int id) {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.